src/MapboxStyleParser.ts

```diff
@@ -0,0 +1,1087 @@
+import {
+    PointSymbolizer,
+    Rule,
+    Style,
+    StyleParser,
+    Symbolizer,
+    FillSymbolizer,
+    LineSymbolizer,
+    IconSymbolizer,
+    TextSymbolizer,
+    Filter,
+    Operator,
+    SymbolizerKind,
+    MarkSymbolizer,
+    ScaleDenominator,
+    UnsupportedProperties
+} from 'geostyler-style';
+
+import MapboxStyleUtil from './Util/MapboxStyleUtil';
+
+const _cloneDeep = require('lodash/cloneDeep');
+const _isEqual = require('lodash/isEqual');
+
+type MapboxLayerType = 'fill' | 'line' | 'symbol' | 'circle' | 'heatmap' |
+    'fill-extrusion' | 'raster' | 'hillshade' | 'background';
+
+type SymbolType = {
+    textSymb?: TextSymbolizer;
+    iconSymb?: IconSymbolizer;
+};
+
+export class MapboxStyleParser implements StyleParser {
+    isSymbolType(s: Symbolizer|SymbolType): s is SymbolType {
+        return (<SymbolType> s).iconSymb ? true : (<SymbolType> s).textSymb ? true : false;
+    }
+
+    /**
+     * Object of unsupported properties.
+     */
+    static unsupportedProperties: UnsupportedProperties = {
+        Symbolizer: {
+            FillSymbolizer: {
+                outlineWidth: 'unsupported',
+                outlineDasharray: 'unsupported',
+                graphicFill: 'unsupported',
+            },
+            LineSymbolizer: {
+                dashOffset: 'unsupported'
+            },
+            MarkSymbolizer: 'unsupported',
+            IconSymbolizer: 'unsupported'
+        }
+    };
+
+    /**
+     * Parses the GeoStylerStyle-SymbolizerKind from a Mapbox Style Layer
+     *
+     * @param {any} layer A Mapbox Style Layer
+     * @return {SymbolizerKind} A GeoStylerStyle-SymbolizerKind
+     */
+    getSymbolizerKindFromMapboxLayer(type: string): SymbolizerKind|'Symbol' {
+        switch (type) {
+            case 'fill':
+                return 'Fill';
+            case 'line':
+                return 'Line';
+            case 'symbol':
+                return 'Symbol';
+            default:
+                throw new Error(`Could not parse mapbox style. Unsupported layer type.`);
```

Comment:
…We support types `fill`, `line` and `symbol` only.

Reply:
Added in https://github.com/terrestris/geostyler-mapbox-parser/pull/1/commits/b8912d60e00672bff2a78e67b869744c5cb9f8b6